dbt/adapters/athena/impl.py

```diff
@@ -514,51 +525,52 @@ def list_relations_without_caching(self, schema_relation: AthenaRelation) -> Lis
         return relations
 
     @available
-    def get_table_type(self, db_name, table_name) -> TableType:
+    def get_table_type(self, relation: AthenaRelation) -> Optional[TableType]:
         conn = self.connections.get_thread_connection()
         client = conn.handle
 
         with boto3_client_lock:
             glue_client = client.session.client("glue", region_name=client.region_name, config=get_boto3_config())
 
         try:
-            response = glue_client.get_table(DatabaseName=db_name, Name=table_name)
-            _type = self.relation_type_map.get(response.get("Table", {}).get("TableType"))
-            _specific_type = response.get("Table", {}).get("Parameters", {}).get("table_type", "")
+            response = glue_client.get_table(DatabaseName=relation.schema, Name=relation.identifier)
+        except glue_client.exceptions.EntityNotFoundException as e:
+            logger.debug(f"Error calling Glue get_table: {e}")
+            return None
```

Comment:
shouldn't we raise an error instead of returning a None Downstream?

Reply:
Hum I think I can clean things up a little because we recall the get_table_type after getting already the table which makes the call unnecessary ...